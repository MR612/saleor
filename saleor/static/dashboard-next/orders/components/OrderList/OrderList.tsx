import Card from "material-ui/Card";
import blue from "material-ui/colors/blue";
import IconButton from "material-ui/IconButton";
import { withStyles } from "material-ui/styles";
import Table, {
  TableBody,
  TableCell,
  TableFooter,
  TableHead,
  TableRow
} from "material-ui/Table";
import * as React from "react";

import Skeleton from "../../../components/Skeleton";
import TablePagination from "../../../components/TablePagination";
import i18n from "../../../i18n";

interface OrderListProps {
  orders?: Array<{
    id: string;
    number: number;
    status: string;
    client: {
      id: string;
      email: string;
    };
    created: string;
    paymentStatus: string;
    price: {
      localized;
    };
  }>;
  hasPreviousPage?: boolean;
  hasNextPage?: boolean;
  onPreviousPage?();
  onNextPage?();
  onRowClick?(id: string);
}

const decorate = withStyles(theme => ({
  link: {
    color: blue[500],
    cursor: "pointer",
    textDecoration: "none"
  },
  textRight: {
    textAlign: "right" as "right"
  }
}));
const OrderList = decorate<OrderListProps>(
  ({
    classes,
    orders,
    hasPreviousPage,
    onPreviousPage,
    hasNextPage,
    onNextPage,
    onRowClick
  }) => (
    <Card>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>{i18n.t("#", { context: "object" })}</TableCell>
            <TableCell>{i18n.t("Status", { context: "object" })}</TableCell>
            <TableCell>{i18n.t("Client", { context: "object" })}</TableCell>
            <TableCell>{i18n.t("Created at", { context: "object" })}</TableCell>
            <TableCell>{i18n.t("Payment", { context: "object" })}</TableCell>
            <TableCell className={classes.textRight}>
              {i18n.t("Price", { context: "object" })}
            </TableCell>
          </TableRow>
        </TableHead>
        <TableFooter>
          <TableRow>
            <TablePagination
              colSpan={6}
              hasNextPage={hasNextPage || false}
              onNextPage={onNextPage}
              hasPreviousPage={hasPreviousPage || false}
              onPreviousPage={onPreviousPage}
            />
          </TableRow>
        </TableFooter>
        <TableBody>
          {orders === undefined || orders === null ? (
            <TableRow>
              <TableCell>
                <Skeleton />
              </TableCell>
              <TableCell>
                <Skeleton />
              </TableCell>
              <TableCell>
                <Skeleton />
              </TableCell>
              <TableCell>
                <Skeleton />
              </TableCell>
              <TableCell>
                <Skeleton />
              </TableCell>
              <TableCell>
                <Skeleton />
              </TableCell>
            </TableRow>
          ) : orders.length > 0 ? (
            orders.map(order => (
              <TableRow key={order.id}>
                <TableCell
                  onClick={onRowClick ? onRowClick(order.id) : () => {}}
                  className={classes.link}
                >
                  {order.number}
                </TableCell>
                <TableCell>{order.status}</TableCell>
                <TableCell>{order.client.email}</TableCell>
                <TableCell>{order.created}</TableCell>
                <TableCell>{order.paymentStatus}</TableCell>
                <TableCell className={classes.textRight}>
                  {order.price.localized}
                </TableCell>
              </TableRow>
            ))
          ) : (
            <TableRow>
              <TableCell colSpan={6}>{i18n.t("No orders found")}</TableCell>
            </TableRow>
          )}
        </TableBody>
      </Table>
    </Card>
  )
);
export default OrderList;
