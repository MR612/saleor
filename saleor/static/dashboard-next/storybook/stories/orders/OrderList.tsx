import { storiesOf } from "@storybook/react";
import * as React from "react";

import OrderList from "../../../orders/components/OrderList";
import { orders } from "../../../orders/fixtures";

storiesOf("orders / OrderList", module)
  .add("when loading data", () => <OrderList />)
  .add("when loaded data", () => (
    <OrderList orders={orders.edges.map(edge => edge.node)} />
  ))
  .add("when no data", () => <OrderList orders={[]} />);
