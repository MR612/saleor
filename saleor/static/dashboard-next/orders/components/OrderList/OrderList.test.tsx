import * as React from "react";
import * as renderer from "react-test-renderer";

import OrderList from "./";

describe("<OrderList />", () => {
  it("renders", () => {
    const component = renderer.create(<OrderList />);
    expect(component).toMatchSnapshot();
  });
});
