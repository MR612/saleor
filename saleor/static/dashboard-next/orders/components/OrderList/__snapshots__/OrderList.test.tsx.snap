// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> renders 1`] = `
<div
  className="MuiPaper-root-7 MuiPaper-elevation2-11 MuiPaper-rounded-8 MuiCard-root-6"
>
  <table
    className="MuiTable-root-34"
  >
    <thead
      className="MuiTableHead-root-35"
    >
      <tr
        className="MuiTableRow-root-36 MuiTableRow-head-39"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-head-42"
          scope="col"
        >
          #
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-head-42"
          scope="col"
        >
          Status
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-head-42"
          scope="col"
        >
          Client
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-head-42"
          scope="col"
        >
          Created at
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-head-42"
          scope="col"
        >
          Payment
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-head-42 Component-textRight-4"
          scope="col"
        >
          Price
        </th>
      </tr>
    </thead>
    <tfoot>
      <tr
        className="MuiTableRow-root-36 MuiTableRow-footer-40"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-footer-44 MuiTablePagination-root-52"
          colSpan={6}
          scope={undefined}
        >
          <div
            className="MuiToolbar-root-58 MuiToolbar-gutters-59 MuiTablePagination-toolbar-57"
          >
            <div
              className="MuiTablePagination-spacer-56"
            />
            <div
              className="MuiTablePaginationActions-root-60"
            >
              <button
                className="MuiButtonBase-root-67 MuiButtonBase-disabled-68 MuiIconButton-root-61 MuiIconButton-disabled-65"
                disabled={true}
                onBlur={[Function]}
                onClick={undefined}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="MuiIconButton-label-66"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-69"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                className="MuiButtonBase-root-67 MuiButtonBase-disabled-68 MuiIconButton-root-61 MuiIconButton-disabled-65"
                disabled={true}
                onBlur={[Function]}
                onClick={undefined}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="MuiIconButton-label-66"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-69"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
    <tbody
      className="MuiTableBody-root-75"
    >
      <tr
        className="MuiTableRow-root-36"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-body-43"
          scope={undefined}
        >
          <span
            className="Component-skeleton-76"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-body-43"
          scope={undefined}
        >
          <span
            className="Component-skeleton-76"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-body-43"
          scope={undefined}
        >
          <span
            className="Component-skeleton-76"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-body-43"
          scope={undefined}
        >
          <span
            className="Component-skeleton-76"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-body-43"
          scope={undefined}
        >
          <span
            className="Component-skeleton-76"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-41 MuiTableCell-body-43"
          scope={undefined}
        >
          <span
            className="Component-skeleton-76"
            style={undefined}
          >
            ‌
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
